### DISCUSSION TITLE
Remove old Programs and Variables

### DISCUSSION THREAD

**Discussion by josch88** [SukramJ/aiohomematic#1795] [Q&A] [answered]
Hi,
after deleting several programs and variables in my ccu3, I still see them as deactivated controls or sensors in the homematic ip integration in home assistant.
Examples:
button.ccu3_p_beschattung
binary_sensor.ccu3_sv_nachlicht 

How can I can clean up the integration from these old data?

**Comment by SukramJ** [comment, marked as answer]
Aktivieren -> Neustarten -> Löschen.

Hauptsache Du postest Die richtige Antwort, wenn es funktioniert hat, in den anderen Foren, wo Du die Frage auch gestellt hast. 
Ansonsten sind diese Mehrfachposts nervig und Zeitverschwendung für die Beteiligten.

**Reply by josch88** [reply]
Danke.
Dann sieh das doch an dieser Stelle mal als eine Anregung in die Homematic Integration eine Clean Up Funktion für einen solchen Zweck zu integrieren. Würde Sinn ergeben eine Möglichkeit zu haben nicht mehr existierende CCU Objekte in HA zu löschen.

**Reply by SukramJ** [reply]
Wenn das ginge, dann gäbe es die Funktion.
Deaktivierte "gehören" nicht der Integration.
